modules/mapper-extras/src/main/java/org/elasticsearch/index/mapper/extras/SourceConfirmedTextQuery.java

```diff
@@ -244,8 +244,13 @@ public Weight createWeight(IndexSearcher searcher, ScoreMode scoreMode, float bo
                     termStats.add(new TermStatistics(term.bytes(), 1, 1L));
                 }
             }
-            simScorer = searcher.getSimilarity().scorer(boost, collectionStatistics, termStats.toArray(TermStatistics[]::new));
-            approximationWeight = searcher.createWeight(approximate(in), ScoreMode.COMPLETE_NO_SCORES, 1f);
+            if (termStats.size() > 0) {
+                simScorer = searcher.getSimilarity().scorer(boost, collectionStatistics, termStats.toArray(TermStatistics[]::new));
+                approximationWeight = searcher.createWeight(approximate(in), ScoreMode.COMPLETE_NO_SCORES, 1f);
+            } else {
+                simScorer = null;
```

Comment:
This change was necessary because the test now randomly wraps the similarity into an AssertingSimilarity which [checks that the termStats array is not empty](https://github.com/apache/lucene/blob/main/lucene/test-framework/src/java/org/apache/lucene/tests/search/similarities/AssertingSimilarity.java#L55). This is not the case for the last part in SourceConfirmedTextQueryTests#testTerm, which is avoided by this additional condition.